datafusion/functions/src/core/coalesce.rs

```diff
@@ -60,12 +60,16 @@ impl ScalarUDFImpl for CoalesceFunc {
     }
 
     fn return_type(&self, arg_types: &[DataType]) -> Result<DataType> {
-        Ok(arg_types[0].clone())
+        Ok(arg_types
+            .iter()
+            .find_or_first(|d| !d.is_null())
+            .unwrap()
```

Comment:
I wonder if it is ever possible for this unwrap to panic? I could trigger it by explicitly calling `return_type(&[])` but perhaps this would never happen via SQL

Reply:
Only if the `arg_types` is empty, the unwrap panics. But the length check is handled in `coerce_types`, so it is safe to unwrap here.